httpx/_client.py

```diff
@@ -758,7 +766,7 @@ def request(
         params: QueryParamTypes = None,
         headers: HeaderTypes = None,
         cookies: CookieTypes = None,
-        auth: typing.Union[AuthTypes, UnsetType] = UNSET,
+        auth: typing.Union[AuthTypes, ClientDefaultType] = USE_CLIENT_DEFAULT,
         allow_redirects: bool = True,
         timeout: typing.Union[TimeoutTypes, UnsetType] = UNSET,
```

Comment:
So, *these* cases in `_client.py` should be:

```python
    timeout: typing.Union[TimeoutTypes, ClientDefaultType] = USE_CLIENT_DEFAULT,
```

(But the signature of the arguments in the Timeout *class* should keep using `UNSET`)

Reply:
Actually thinking about this, it's probably also a good idea to move `ClientDefaultType` and `USE_CLIENT_DEFAULT` *out* of `_types.py`, and into this module, to keep them close to where they're actually used.

Reply:
> So, _these_ cases in `_client.py` should be:
> 
> ```python
>     timeout: typing.Union[TimeoutTypes, ClientDefaultType] = USE_CLIENT_DEFAULT,
> ```
> 
> (But the signature of the arguments in the Timeout _class_ should keep using `UNSET`)

If I do that I need to also change the the types in the `Timeout` class, otherwise `mypy` complains.

Reply:
Okay, I'll take a look.